### DISCUSSION TITLE
best JVM arguments for the mod

### DISCUSSION THREAD

**Discussion by c4mp3r-87** [HbmMods/Hbm-s-Nuclear-Tech-GIT#1303] [Q&A]
i need some help with the JVM to run the mod smoothly, i already copied some commands from youtube, but i think there may be better commands to use.

the current JVM i use are:
-XX:+UseG1GC -XX:+UnlockExperimentalVMOptions -Xmx8G -Xms8G -XX:+DisableExplicitGC -XX:ParallelGCThreads=12 -XX:+UseNUMA -Dfml.ignorePatchDiscrepancies=true -Dfml.ignoreInvalidMinecraftCertificates=true -XX:MaxGCPauseMillis=50 -XX:SurvivorRatio=8 -XX:TargetSurvivorRatio=90 -XX:MaxTenuringThreshold=15 -XX:SoftRefLRUPolicyMSPerMB=6000 -XX:ReservedCodeCacheSize=1024m -XX:+UseCodeCacheFlushing -Dorg.lwjgl.opengl.Window.undecorated=true

my PC specs:
-32GB ram
-AMD ryzen 5 4600G with Radeon Graphics

My FPS varies between 30 ~ 120(standing still) , and sometimes i have some huge LAG spikes 
i don't use optifine as it cause some strange bugs sometimes

in game i set all the graphic configs to fancy and render distance on 8

![2024-01-19_13 51 29](https://github.com/HbmMods/Hbm-s-Nuclear-Tech-GIT/assets/157136251/62b37b45-a0cf-4003-9c72-fd6102f945f3)

**Comment by JameH2** [comment]
How many mods do you have

**Reply by c4mp3r-87** [reply]
11
-better FPS
-code chicken core
-code chicken lib
-HBM NTM
-inventory tweaks
-keeping inventory
-NaturesCompass
-NotEnoughItems
-NotenoughWands
-OreExcavation
-Waystones

**Comment by JameH2** [comment]
Actually scratch that, 8 gb is a bit too much because after a certain point java gc will start giving diminishing returns and start performing worse

**Reply by c4mp3r-87** [reply]
so, 4Gb is enough?
